packages/cms-cli/commands/init.js

```diff
@@ -34,83 +34,101 @@ const {
+  trackCommandUsage(COMMAND_NAME, { authType: 'personalaccesskey' });
+  const env = options.qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
 
-      const configPath = getConfigPath();
-      const env = options.qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
+  if (configPath) {
+    logger.error(`The config file '${configPath}' already exists.`);
+    logger.info(
+      'To update an existing config file, use the "hs auth" command.'
+    );
+    process.exit(1);
+  }
 
-      if (configPath) {
-        logger.error(`The config file '${configPath}' already exists.`);
-        logger.info(
-          'To update an existing config file, use the "hs auth" command.'
-        );
-        process.exit(1);
-      }
+  trackAuthAction(
+    COMMAND_NAME,
+    PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
+    TRACKING_STATUS.STARTED
+  );
 
-      trackAuthAction(
-        COMMAND_NAME,
-        PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
-        TRACKING_STATUS.STARTED
-      );
+  createEmptyConfigFile();
+  handleExit(deleteEmptyConfigFile);
 
-      createEmptyConfigFile();
-      handleExit(deleteEmptyConfigFile);
+  try {
+    const configData = await personalAccessKeyPrompt({ env });
+    const { name } = await promptUser([PORTAL_NAME]);
 
-      try {
-        const configData = await personalAccessKeyPrompt({ env });
-        const { name } = await promptUser([PORTAL_NAME]);
+    await updateConfigWithPersonalAccessKey(
+      {
+        ...configData,
+        name,
+      },
+      true
+    );
 
-        await updateConfigWithPersonalAccessKey(
-          {
-            ...configData,
-            name,
-          },
-          true
-        );
+    const path = getConfigPath();
+    const portalId = getPortalId();
 
-        const path = getConfigPath();
-        const portalId = getPortalId();
+    logger.success(
+      `The config file "${path}" was created using your personal access key for portal ${portalId}.`
+    );
 
-        logger.success(
-          `The config file "${path}" was created using your personal access key for portal ${portalId}.`
-        );
+    trackAuthAction(
+      COMMAND_NAME,
+      PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
+      TRACKING_STATUS.COMPLETE,
+      portalId
+    );
+  } catch (err) {
+    logErrorInstance(err);
+    trackAuthAction(
+      COMMAND_NAME,
+      PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
+      TRACKING_STATUS.ERROR
+    );
+  }
+};
 
-        trackAuthAction(
-          COMMAND_NAME,
-          PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
-          TRACKING_STATUS.COMPLETE,
-          portalId
-        );
-      } catch (err) {
-        logErrorInstance(err);
-        trackAuthAction(
-          COMMAND_NAME,
-          PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
-          TRACKING_STATUS.ERROR
-        );
-      }
-    });
+// Yargs Configuration
+const command = `${COMMAND_NAME}`;
+const describe = DESCRIPTION;
+const builder = yargs => {
+  addLoggerOptions(yargs, true);
+  addTestingOptions(yargs, true);
```

Comment:
do we want `addHelpUsageTracking` here as well?

Reply:
Yeah I totally would, but I can't find a way to make it work with args.  Commander exposes an `on('--help', () =>` method, but I can't find an equivalent for Yargs.

Reply:
Yeah I see what you mean about there not being a clear event for this in yargs to attach to.

Currently, we display help in two cases `--help` and if the arguments provided are not enough for the command/subcommand.

My thought is perhaps we can utilize [.middleware](https://yargs.js.org/docs/#api-reference-middlewarecallbacks-applybeforevalidation) to check the arguments provided are enough for the command, similarly to how we are currently doing it([create command example](https://github.com/HubSpot/hubspot-cms-tools/blob/49748935b7b965b28a8df2280749abb408386538/packages/cms-cli/bin/hs-create.js#L11)) and also check for the presence of `--help` option.

The tricky part will be that different commands require different arguments.

Reply:
I'd also prefer it if we moved away from using an option for `--qa`, which is a feature that is only for HubSpotters especially if we can't suppress it from the help. With `yargs`, can we hide an option from help?

Reply:
Yep, there's a `hidden` prop, I'll add it